fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreBase.java

```diff
@@ -299,13 +299,34 @@ public boolean errorIfTypeChanged() {
     /**
      * Provided during record save (via {@link #saveRecord(Message, FDBRecordVersion, VersionstampSaveBehavior)}),
      * directs the behavior of the save w.r.t. the record's version.
-     * In the presence of a version, either <code>DEFAULT</code> or <code>WITH_VERSION</code> can be used.
+     * In the presence of a version, either {@code DEFAULT} or {@code WITH_VERSION} can be used.
      * For safety, <code>NO_VERSION</code> should only be used with a null version.
      */
     enum VersionstampSaveBehavior {
-        DEFAULT,        // Follow rules dictated by the metadata
-        NO_VERSION,     // Explicitly do NOT save a version
-        WITH_VERSION,   // Explicitly save a version even if meta-data says not to
+        /**
+         * Match the behavior dictated by the meta-data. If {@link com.apple.foundationdb.record.RecordMetaData#isStoreRecordVersions()}
+         * returns {@code true}, this will always store the record with a version (like {@link #WITH_VERSION}). Otherwise,
+         * it will store the record with the provided version if given or with no version if not.
```

Comment:
Yeah, that sounds like a good idea. I added it to `WITH_VERSION` instead of `DEFAULT`, as I think it applies to both and `DEFAULT` already references `WITH_VERSION`